tools/project.py

```diff
@@ -1388,6 +1414,253 @@ def unix_path(input: Any) -> str:
+            def try_replace(flag: str) -> bool:
+                if flag in CFLAG_REPLACE:
+                    cflags.extend(CFLAG_REPLACE[flag])
+                    return True
+
+                for prefix, replacement in CFLAG_REPLACE_PREFIX:
+                    if flag.startswith(prefix):
+                        cflags.append(flag.replace(prefix, replacement, 1))
+                        return True
+
+                return False
+
+            for flag in flags:
+                # Ignore flags first
+                if should_ignore(flag):
+                    continue
+
+                # Then find replacements
+                if try_replace(flag):
+                    continue
+
+                # Pass flags through last
+                if should_passthrough(flag):
+                    cflags.append(flag)
+                    continue
+
+        # Since we need to do processing on individual flags, cflag strings are not supported
+        if not isinstance(obj.options["cflags"], list):
+            print(
+                f"clangd config generation does not support cflags as a single string: {obj.src_path}"
+            )
+            return
+        append_cflags(obj.options["cflags"])
+
+        if isinstance(obj.options["extra_cflags"], list):
+            append_cflags(obj.options["extra_cflags"])
+
+        mw_version = Path(obj.options["mw_version"])
+        mwcc = compilers / mw_version / "mwcceppc.exe"
+
+        # Apply compiler defines, if requested
+        if config.clangd_compiler_defines:
+            mw_version_str = mw_version.as_posix()
+            compiler_defines = COMPILER_DEFINES.get(mw_version_str)
+            if compiler_defines is None:
+                print(f"Missing version value for compiler {mw_version_str}")
+            else:
+                for define in compiler_defines:
+                    cflags.append(f"-D{define}")
+
+        in_path = obj.src_path
+        out_path = obj.src_obj_path
+
+        unit_config = {
+            "directory": Path.cwd(),
+            "file": in_path,
+            "output": out_path,
+            # "command": mwcc_cmd,
+            "arguments": [
+                *([wrapper, mwcc] if wrapper else [mwcc]),
+                *cflags,
+                "-MMD",
+                "-c",
+                in_path,
+                "-o",
+                out_path.parent,
+            ],
+        }
+        clangd_config.append(unit_config)
+
+    # Add DOL units
+    for unit in build_config["units"]:
+        add_unit(unit)
+
+    # Add REL units
+    for module in build_config["modules"]:
+        for unit in module["units"]:
+            add_unit(unit)
+
+    # Write compile_commands.json
+    with open("compile_commands.json", "w", encoding="utf-8") as w:
+
+        def default_format(o):
+            if isinstance(o, Path):
+                return o.resolve().as_posix()
+            return str(o)
+
+        json.dump(clangd_config, w, indent=4, default=default_format)
```

Comment:
```suggestion
        json.dump(clangd_config, w, indent=2, default=default_format)
```
I just recently changed the others to indent=2, not that it matters much